cfg/builder/builder_walk.cc

```diff
@@ -223,22 +223,23 @@ BasicBlock *CFGBuilder::walkHash(CFGContext cctx, ast::Hash &h, BasicBlock *curr
     LocalRef magic = cctx.newTemporary(core::Names::magic());
     InlinedVector<cfg::LocalRef, 2> vars;
     InlinedVector<core::LocOffsets, 2> locs;
+    auto isPrivateOk = false;
+    auto numArgs = 2 * h.keys.size();
+    auto snd = Send::make(magic, h.loc, method, core::LocOffsets::none(), numArgs, isPrivateOk, numArgs);
+
     for (auto [key, val] : h.kviter()) {
         LocalRef keyTmp = cctx.newTemporary(core::Names::hashTemp());
         LocalRef valTmp = cctx.newTemporary(core::Names::hashTemp());
         current = walk(cctx.withTarget(keyTmp), key, current);
         current = walk(cctx.withTarget(valTmp), val, current);
-        vars.emplace_back(keyTmp);
-        vars.emplace_back(valTmp);
-        locs.emplace_back(key.loc());
-        locs.emplace_back(val.loc());
+        *snd.refs++ = keyTmp;
+        *snd.refs++ = valTmp;
+        *snd.locs++ = key.loc();
+        *snd.locs++ = val.loc();
```

Comment:
Is the API set up to make sure that you can't use this pattern at some later point to attempt to mutate the args after construction?

Reply:
Or alternatively, should we have a helper function where you can pass a list in, so that there's no possibility of you messing up the `numArgs` yourself?